hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/placement/algorithms/TestSCMContainerPlacementRackAware.java

```diff
@@ -82,7 +82,7 @@ public void setup() {
     when(nodeManager.getNodeStat(anyObject()))
         .thenReturn(new SCMNodeMetric(STORAGE_CAPACITY, 0L, 100L));
     when(nodeManager.getNodeStat(datanodes.get(2)))
-        .thenReturn(new SCMNodeMetric(STORAGE_CAPACITY, 90L, 10L));
+        .thenReturn(new SCMNodeMetric(STORAGE_CAPACITY, 90L, 20L));
```

Comment:
Thanks @ChenSammi  for the patch. I had though about similar test only fix yesterday. 

But this may hide a code bug when we deal with mixed nodes that some have enough space while others don't. If the chooseNodes() keep choosing the nodes without enough capacity for 3 times (default), we end up with failures even though other nodes have enough space. Here are two proposals:

1. Filtering the candidate node list without enough capacity so that the placement algorithm won't need to deal with it.

2. Handle capacity in the placement algorithm by adding the detected node without enough capacity to the exclude node list, so that the algorithm won't choose them in the next round. 

I prefer 1 because it seems a cleaner approach and more efficient than 2. I'm OK with 2 as well if you feel it is easier to adjust the placement algorithm.

Reply:
@xiaoyuyao, currently I favor 2. The chooseRandom node logic in NetworktopologyImpl now only has the information of toplogy. It cannot make decision which node has enough capacity since it doesn't have the knowledge of disk free space so far. So I will go with solution 2 now.